dissect/target/plugins/apps/remoteaccess/teamviewer.py

```diff
@@ -45,82 +70,140 @@ class TeamViewerPlugin(RemoteAccessPlugin):
     def __init__(self, target):
         super().__init__(target)
 
-        self.logfiles: list[list[TargetPath, UserDetails]] = []
+        self.logfiles: set[tuple[str, UserDetails | None]] = set()
 
         # Find system service log files.
         for log_glob in self.SYSTEM_GLOBS:
             for logfile in self.target.fs.glob(log_glob):
-                self.logfiles.append([logfile, None])
+                self.logfiles.add((logfile, None))
 
         # Find user log files.
         for user_details in self.target.user_details.all_with_home():
             for log_glob in self.USER_GLOBS:
                 for logfile in user_details.home_path.glob(log_glob):
-                    self.logfiles.append([logfile, user_details])
+                    self.logfiles.add((logfile, user_details))
 
     def check_compatible(self) -> None:
         if not len(self.logfiles):
-            raise UnsupportedPluginError("No Teamviewer logs found")
+            raise UnsupportedPluginError("No Teamviewer logs found on target")
 
     @export(record=RemoteAccessLogRecord)
     def logs(self) -> Iterator[RemoteAccessLogRecord]:
         """Yield TeamViewer client logs.
 
         TeamViewer is a commercial remote desktop application. An adversary may use it to gain persistence on a system.
         """
+
+        try:
+            target_tz = self.target.datetime.tzinfo
+
+        except Exception as e:
+            self.target.log.warning("Unable to determine target timezone, assuming UTC if no timezone is found in logs")
+            self.target.log.debug("", exc_info=e)
+            target_tz = timezone.utc
+
         for logfile, user_details in self.logfiles:
             logfile = self.target.fs.path(logfile)
 
             start_date = None
-            with logfile.open("rt") as file:
+            with logfile.open("rt") as fh:
                 while True:
                     try:
-                        line = file.readline()
-                    except UnicodeDecodeError:
+                        line = fh.readline()
+                    except UnicodeDecodeError as e:
+                        self.target.log.warning("Unable to parse log line in %s: %s", logfile, e)
+                        self.target.log.debug("", exc_info=e)
                         continue
 
-                    # End of file, quit while loop
                     if not line:
                         break
 
                     line = line.strip()
```

Comment:
```suggestion
```
Can just let it exhaust the iterator.

Reply:
Can you add a unit test for why that manual `fh.readline()` is necessary?

Reply:
The `while True` can definitely be replaces with just a `for line in fh:`. Please go ahead and 😄

Reply:
Well, the `while True` is used to skip lines that cause a decoding error. In that way, we only skip a line, instead of the full remainder of the file.

We could add `errors="replace"` in the `with` statement, to replace the decoding errors in the lines and we just continue parsing that line with the replacements. Then we don't need the `while`-statement